### PR TITLE
nc-news-7: create POST /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by sophocles99**

### CODE REVIEW

File: __tests__/articles.test.js [hunk 1]
```diff
@@ -122,3 +121,88 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe.only("POST: /api/articles/:article_id/comments", () => {
+  test("201: returns the successfully posted comment", () => {
```
Comment (MKD01) [inline __tests__/articles.test.js]: Please dont forget to remove any .only or .skip's that you have.

**Review by MKD01 — CHANGES_REQUESTED** [review]
Please dont push any code with .only or .skip's as you could be pushing broken tests/code without knowing this. Once you have removed all of these please resubmit this pr.

File: __tests__/articles.test.js [hunk 2]
```diff
@@ -122,3 +121,88 @@ describe("GET /api/articles/:article_id/comments", () => {
...
 });
+
+describe("POST: /api/articles/:article_id/comments", () => {
+  test("201: returns the successfully posted comment", () => {
+    const testComment = {
+      comment: {
+        username: "butter_bridge",
+        body: "Bemused of Basingstoke. What is this guy on?",
+      },
+    };
```
Comment (johnhutcheon) [inline __tests__/articles.test.js:132]: Can we add another 201 test here please? This should ignore unnecessary properties, i.e. if something other than username and body is sent.

File: __tests__/articles.test.js [hunk 3]
```diff
@@ -122,3 +121,88 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+        const { comment } = body;
+        expect(comment).toHaveProperty("comment_id", expect.any(Number));
+        expect(comment).toHaveProperty("body", expect.any(String));
+        expect(comment).toHaveProperty("article_id", expect.any(Number));
+        expect(comment).toHaveProperty("author", expect.any(String));
+        expect(comment).toHaveProperty("votes", 0);
+        expect(comment).toHaveProperty("created_at", expect.any(String));
+      });
+  });
+  test("400: returns error if article_id doesn't exist", () => {
```
Comment (johnhutcheon) [inline __tests__/articles.test.js]: We would expect a 404 here not 400 👍

File: __tests__/articles.test.js [hunk 4]
```diff
@@ -122,3 +121,88 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+    };
+    return request(app)
+      .post("/api/articles/14/comments")
+      .send(testComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
+  test("400: returns error if username doesn't exist", () => {
```
Comment (johnhutcheon) [inline __tests__/articles.test.js]: And this one should also be 404 please!

File: __tests__/articles.test.js [hunk 5]
```diff
@@ -122,3 +121,88 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+      },
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(testComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
```
Comment (johnhutcheon) [inline __tests__/articles.test.js:225]: Could we just add one more 400 test please for invalid ID. I.e. articles/bananas/comments ?

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hey David, really nice work so far! Your testing is really thorough, but I've just requested a couple of extra ones and some amendments if you could do this please. Thank you! 🙂

**Review by yalicee — APPROVED** [review]
Good changes happy for you to merge

**Pull request merged**